### PR TITLE
Sudoku solver merge

### LINKED ISSUE #6: Sudoku Solver

### PR DESCRIPTION
**Pull request by Mathieufalardeau3**
This is the code for the Sudoku Solver. Take a look and review it.

### CODE REVIEW

**Review by gabriels2018 — APPROVED** [review]
I approve the merge request

File: test/src/solver_sudoku_test.c [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+#define ROWS 9
+#define COLS 9
+
+/**
+* This function sorts the 3x3 values and test them if they pass the Sudoku rules
+* @param[in] tester[ROWS] contains either the row, columns or the 3x3 solver output values.
+* @param[out] _Bool is used to check if there are duplicate values in either row, column or
+*             the 3x3 solver output, if there are duplicates it returns false otherwise true.
+*/
+ _Bool sortArray_and_test(int tester[ROWS]){
```
Comment (Mathieufalardeau3) [inline test/src/solver_sudoku_test.c]: Make sure to follow the naming convention.

**Review by Prasanthkota165 — CHANGES_REQUESTED** [review]
Can you guys do this explanation in SUDOKU.H?

File: src/solver_sudoku.c [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+#define N 9 /*A constant for the basic size of a Sudoku*/
+
+/**
+ * Determines if the sudoku puzzle is solvable. 
+ * 
+ * @param[in] sudoku_puzzle[N][N] The Sudoku puzzle to be verified.
+ * @param[out] temp_sudoku[N][N] The Sudoku puzzle to be verified.
+ * 
+ * @return TRUE if it is solvable, FASLE if it is not.  
+ */
```
Comment (Prasanthkota165) [inline src/solver_sudoku.c:35]: This part.

File: src/solver_sudoku.c [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+}
+
+/**
+ * Recursive function that applies the main mechanic to solve the Sudoku 
+ * puzzle. 
+ * 
+ * @param[in] sudoku_puzzle[N][N] A partially solved Sudoku puzzle.
+ * 
+ * @return TRUE if the Sudoku Puzzle has been solved, FALSE if it 
+ */
```
Comment (Prasanthkota165) [inline src/solver_sudoku.c:63]: Can you include in SUDOKU.H?

**Review by rishithrao57 — APPROVED** [review]
Looks good.

**Review by Prasanthkota165 — APPROVED** [review]

**Review by gabriels2018 — APPROVED** [review]
I approve the merge to development

**Pull request merged**